File: itv/SergioDeIscar-itv-java/SergioDeIscar-itv-java/src/main/java/org/example/interfaces/ItvExtension.java
package org.example.interfaces;

import org.example.models.Vehiculo;

import java.util.List;
import java.util.Map;

public interface ItvExtension extends CrudRepository<Vehiculo, String>{
    Boolean isFull();
    List<Vehiculo> getCoches();
    Vehiculo getCocheMaxModerno();
    Vehiculo getVehiculoMinKilometros();
    List<Vehiculo> findByMarca(String marca);
    Double getMediaKilometrosMotos();
    Vehiculo getOldestCoche();
    Vehiculo getOldestCoche2Puertas();
    Map<String, Integer> getCountVehiculos();
    Map<String, Integer> getCountVehiculosAptos();
    Map<String, Double> getMediaVehiculosAnio();
    Map<String, List<Vehiculo>> groupVehiculosByMarca();
    List<Vehiculo> sortVehiculosByAnio();
    List<Vehiculo> sortDesVehiculosByMarca();
    Map<String, List<Vehiculo>> groupVehiculosByMarcaSortByKilometro();
}

File: itv/SergioDeIscar-itv-kotlin/SergioDeIscar-itv-kotlin/src/main/kotlin/interfaces/ItvExtension.kt
package interfaces

import models.Vehiculo

interface ItvExtension: CrudRepository<Vehiculo, String> {
    fun isFull():Boolean
    fun getCoches(): List<Vehiculo>
    fun getCocheMaxModerno(): Vehiculo
    fun getVehiculoMinKilometros(): Vehiculo
    fun getMediaKilometrosMotos(): Double
    fun getOldestCoche(): Vehiculo
    fun getCountVehiculos(): Map<String, Int>
    fun getCountVehiculosAptos(): Map<String, Int>
    fun getMediaVehiculosAnio(): Map<String, Double>
    fun groupVehiculosByMarca(): Map<String, List<Vehiculo>>
    fun sortVehiculosByAnio(): List<Vehiculo>
    fun sortDesVehiculosByMarca(): List<Vehiculo>
    fun groupVehiculosByMarcaSortByKilometro(): Map<String, List<Vehiculo>>
}